### PR TITLE
Preparing support for self-hosting for real

### PR DESCRIPTION
**Pull request by confact**
Adding more env variables to be able to set everything to own host. All on the backend is fixed.

The last thing is the javascript.
Need to find a good way to set the domain in the javascript.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -49,12 +49,9 @@ crystal spec
 And the code should be in `/src` and the tests in `/spec`
 
 ### Deployment
-You can easily deploy this. Just clone this to your computer and run `docker build .` and deploy that container to your server. Kindmetrics use all lucky framework environment variables which can be seen here: https://luckyframework.org/guides/deploying/ubuntu#creating-a-systemd-unit-file-for-your-app
-
-Kindmetrics is dependent on some SMTP mail server and postgresql. Be sure to have those prepared.
+You can find the deployment instructions in [deployment.md file](https://github.com/kindmetrics/kindmetrics/blob/master/deployment.md)
```
Comment (alexanderadam) [inline README.md:52]: Just a detail, but if you use a relative path (`[deployment.md file](/deployment.md)`) you can easily migrate to Git(Lab/tea/whatever) at a later point and still have working links. :wink:

Comment (confact) [reply README.md:52]: Good point. will fix that :)

**Comment by alexanderadam** [comment]
I guess, generating `track.js` once and mounting it via Docker volume afterwards, would also be a possibility.

This is only needed because `track.js` needs `KINDURL` / `kindmetricsURL`, right?

I'm not sure whether this make sense but wouldn't it also be possible to set the URL like this:

```javascript
KINDURL || (function(){
  // if the parameter is passed via query parameter 'KINDURL'
  // for instance track.js?kind_url=https%3A%2F%2Fkindmetrics.io%2Fsome%2Fpath
  var GET = {};

  for (var i = 0, len = scripts.length; i < len; i++) {
    var src = scripts[i].getAttribute('src').split('?');
    var url = src[0];
    var args = src[1];

    if (!args) {
      continue;
    }

    var query = args.split("&");
    for (var i = 0, max = query.length; i < max; i++)
    {
      if (query[i] === "") // check for trailing & with no param
        continue;

      var param = query[i].split("=");
      GET[decodeURIComponent(param[0])] ||= decodeURIComponent(param[1] || "");
    }
  }

  return GET['KINDURL'];
}())
```

This way it would also be possible to set it via query parameter (`track.js?kind_url=https%3A%2F%2Fkindmetrics.io%2Fsome%2Fpath`).

I'm not sure whether this would be a good idea or whether this would even work though. :wink: 

Either way, you should probably add a sanity check in `track.js`, to ensure that `kindmetricsURL` is set and put a warning to console, right? :thinking: 
Otherwise people might forget setting it (or setting it wrong).

**Comment by confact** [comment]
Good point @alexanderadam - Now it defaults to app.kindmetrics.io. Maybe should not do that and instead force the env variable to be set?

I will think about what is best here. I want this to be as good as possible, so it works well when I publish it on Linode, Heroku, and Digitalocean.

File: src/actions/tracker/render_tracker.cr [hunk 2]
```diff
@@ -1,15 +1,18 @@
-class TrackerJsAction < BrowserAction
+class RenderTracker < BrowserAction
   include Auth::AllowGuests
 
   get "/js/track.js" do
+    if KindEnv.env("APP_HOST").nil?
+      return head 414
```
Comment (alexanderadam) [inline src/actions/tracker/render_tracker.cr:6]: URI Too Long? :thinking:

Comment (confact) [reply src/actions/tracker/render_tracker.cr:6]: @alexanderadam - hum, yea. I read 414 as URI Too Short, maybe reading what I want it to be? :D

`412` or `425` might be better. What do you think?

Comment (alexanderadam) [reply src/actions/tracker/render_tracker.cr:6]: I don't know. Maybe `412`

Comment (confact) [reply src/actions/tracker/render_tracker.cr:6]: I will go with `412` sounded better than 425 as the description of 425 doesn't match as good as 412 does.

**Pull request merged**